src/nvim/shada.c

```diff
@@ -2318,6 +2330,48 @@ static inline void add_search_pattern(PossiblyFreedShadaEntry *const ret_pse,
   }
 }
 
+/// Initialize registers for writing to the ShaDa file
+///
+/// @param[in]  wms  The WriteMergerState used when writing.
+/// @param[in]  max_reg_lines  The maximum number of register lines.
+static void shada_initialize_registers(WriteMergerState *const wms,
+                                       int max_reg_lines)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_ALWAYS_INLINE
+{
+  const void *reg_iter = NULL;
+  const bool limit_reg_lines = max_reg_lines >= 0;
+  do {
+    yankreg_T reg;
+    char name = NUL;
+    bool is_unnamed = false;
+    reg_iter = op_register_iter(reg_iter, &name, &reg, &is_unnamed);
+    if (name == NUL) {
+      break;
+    }
+    if (limit_reg_lines && reg.y_size > max_reg_lines) {
+      continue;
+    }
+    wms->registers[op_reg_index(name)] = (PossiblyFreedShadaEntry) {
+      .can_free_entry = false,
+      .data = {
+        .type = kSDItemRegister,
+        .timestamp = reg.timestamp,
+        .data = {
+          .reg = {
+            .contents = (char **) reg.y_array,
```

Comment:
This would not pass linter either: need no space after a cast. Note that file was written before linter check in question was introduced and I was personally more fond of the space in most cases. (Also note that `(PossiblyFreedShadaEntry) {` above is not a cast, it needs not be changed.)